docs: DOC-853 provide FIPS content (#1496)
spectrocloud/librarium: content/docs/12.5-vertex/00-fips/01-fips-compliant-components.md

| **Component** | **Description** |
| --- | --- |
| CoreDNS | CoreDNS is a DNS server that is deployed as a cluster DNS service. |
| Etcd | Etcd is a distributed key-value store that is used as Kubernetes’ backing store for all cluster data. |
| Metrics Server | Metrics Server is a scalable, efficient source of container resource metrics for Kubernetes built-in autoscaling pipelines. |
| Ingress Controller| Nginx is used as the ingress controller. An ingress controller is a piece of software that provides reverse proxy, configurable traffic routing, and TLS termination for Kubernetes services. |
| Nginx Server| The Nginx server is a web server that can also be used as a reverse proxy, load balancer, mail proxy, and HTTP cache. |
| Nginx Ingress Controller| The Nginx ingress controller is an ingress controller that uses ConfigMap to store the Nginx configuration. |

### Runtime Components

| **Component** | **Description** |
| --- | --- |
| containerd |  Containerd is an industry-standard container runtime with an emphasis on simplicity, robustness, and portability. |
| containerd-shim | Containerd-shim is a shim used by containerd to launch containers. |

> | containerd-shim-runc-v1 | Containerd-shim-runc-v1 is a shim used by containerd to launch containers. |

Reviewer: @karl-cardenas-coding To distinguish between these shims, could we add the part in bold for v1 and v2? 

"Containerd-shim-run-v1 is the **version 1** shim used by ..."
"Containerd-shim-run-v1 is the **version 2** shim used by ..."

I'm just showing it boldface here to indicate the addition, not to make it bold in the publish version.

| containerd-shim-runc-v2 | Containerd-shim-runc-v2 is a shim used by containerd to launch containers. |
| ctr| Ctr is a command line interface for containerd. |
| crictl | Crictl is a command line interface for CRI-compatible container runtimes. |

> | runc | Runc is a CLI tool for spawning and running containers according to the OCI specification. |

Reviewer: ```suggestion
| runc | Runc is a Command Line Interface (CLI) tool for spawning and running containers according to the OCI specification. |
```

Reviewer: "Running" is abelist language.

### Container Network Interface Components

| **Component** | **Description** |
| --- | --- |

> | Calico | Calico is a CNI plugin that provides networking and network policy for Kubernetes clusters. |

Reviewer: Do we spell out CNI here, or not because it's in the heading? Or do we add (CNI) in the heading?

Same for CSI in next section.

> | Calico | Calico is a CNI plugin that provides networking and network policy for Kubernetes clusters. |

Author: ```suggestion
| Calico | Calico is a Container Network Interface plugin that provides networking and network policy for Kubernetes clusters. |
```

| External Snapshotter | The external snapshotter is a CSI plugin that takes snapshots of volumes. |
| Liveness Probe | The liveness probe is a CSI plugin that checks the health of the driver. |
| Node Driver Registrar | The node driver registrar is a CSI plugin that registers the driver with the kubelet. |

<br />

##### Longhorn CSI Components

| **Component**                 | **Description**  |
|---------------------------|--------------|
| Backing image manager     | Manages backing images for Longhorn volumes. |
| Attacher              | Handles attaching and detaching of volumes to nodes. |
| Provisioner           | Manages provisioning and de-provisioning of storage resources. |
| Resizer               | Enables resizing of storage volumes. |
| Snapshotter           | Manages snapshots of Longhorn volumes. |
| Node driver registrar | Registers the CSI driver with the Kubernetes node. |
| Liveness probe            | Monitors health of CSI components. |
| Longhorn engine           | Core component that handles read and write operations to the storage backend. |
| Longhorn instance manager | Manages Longhorn engine and replica instances. |

> | Longhorn share manager    | Manages shared volumes and exposes them via protocols like NFS. |

Reviewer: ```suggestion
| Longhorn share manager    | Manages shared volumes and exposes them via protocols like Network File System (NFS). |
```

| Longhorn UI               | User interface for managing Longhorn components and resources. |
| Longhorn support bundle kit| Collects logs and system information for debugging. |

<InfoBox>

> The Longhorn Manager component is partially FIPS-compliant. This component uses utiltities that are not using a FIPS- compliant version of OpenSSL. The following utilities are not FIPS-compliant:

Reviewer: ```suggestion
The Longhorn Manager component is partially FIPS-compliant. This component uses utiltities that are not using a FIPS-compliant version of OpenSSL. The following utilities are not FIPS-compliant:
```
